http://developer.pidgin.im/ticket/10224
http://developer.pidgin.im/viewmtn/revision/info/37aa00d044431100d37466517568640cb082680c

#
#
# patch "libpurple/protocols/yahoo/libymsg.c"
#  from [2e325ac513bc39fb2cb3964c664edcef344c7f14]
#    to [87ad7ef7612ac888772e8afa7bcb20743f1b3d9c]
#
============================================================
--- libpurple/protocols/yahoo/libymsg.c	2e325ac513bc39fb2cb3964c664edcef344c7f14
+++ libpurple/protocols/yahoo/libymsg.c	87ad7ef7612ac888772e8afa7bcb20743f1b3d9c
@@ -4500,6 +4500,12 @@ void yahoo_set_status(PurpleAccount *acc
 
 	if (purple_presence_is_idle(presence))
 		yahoo_packet_hash_str(pkt, 47, "2");
+	else	{
+		if (!purple_status_is_available(status))
+			yahoo_packet_hash_str(pkt, 47, "1");
+		else
+			yahoo_packet_hash_str(pkt, 47, "0");
+	}
 
 	yahoo_packet_send_and_free(pkt, yd);
 
